src/main/java/org/spongepowered/common/data/SpongeDataManager.java

```diff
@@ -289,147 +298,81 @@ public DataContainer createContainer(DataView.SafetyMode safety) {
         return new MemoryDataContainer(safety);
     }
 
-    void validateRegistration(SpongeDataRegistration registration) {
-        Preconditions.checkState(allowRegistrations);
-
-        // TODO do we want this?
-        Preconditions.checkState(registration.key.getNamespace().equals(registration.plugin.getMetadata().getId()), "Registration namespace not matching plugin id {}", registration.key);
-
-        // Make sure the Keys are not already registered
-        Preconditions.checkState(Collections.disjoint(this.registrationByKey.keySet(), registration.keys), "Duplicate key registration {}", registration.key);
-        Preconditions.checkState(Collections.disjoint(this.registrationByKey.keySet(), registration.dataProviderMap.keySet()), "Duplicate key registration {}", registration.key);
-    }
-
     public static boolean areRegistrationsComplete() {
-        return !allowRegistrations;
+        return !SpongeDataManager.allowRegistrations;
     }
 
     public <E extends DataHolder> void registerKeyListener(KeyBasedDataListener<E> keyListener) {
-        if (areRegistrationsComplete()) { // TODO do we need actually to wait for listener registration?
+        if (SpongeDataManager.areRegistrationsComplete()) { // TODO do we need actually to wait for listener registration?
             this.registerKeyListener(keyListener);
         } else {
             this.keyListeners.add(keyListener);
         }
     }
 
     @SuppressWarnings({"unchecked", "rawtypes"})
-    public void registerDataRegistration(SpongeDataRegistration registration) {
-        this.validateRegistration(registration);
+    public void registerCustomDataRegistration(SpongeDataRegistration registration) {
+        Preconditions.checkState(SpongeDataManager.allowRegistrations);
```

Comment:
Avoid adding more dependencies/additions to using Preconditions as we're moving away from Guava explicitly in several aspects. Also, needs an error message instead of just a random `IllegalStateException`.

Reply:
https://github.com/SpongePowered/SpongeCommon/commit/23cd5efefb8196eb042b8dab6b4bacc03d684226#diff-c702aaa65caf08a9f02f691cb6dd3b97c301e6a2b366cdf22fa2044141012f50R338

![image](https://user-images.githubusercontent.com/1575592/97017867-b8621480-154e-11eb-9931-eb4768db387c.png)